### DISCUSSION TITLE
Rationale for existence of Azure/Modules

### DISCUSSION THREAD

**Discussion by segraef** [Azure/ResourceModules#38] [Q&A]
## Description

In general the question is if it makes sense to develop Azure Modules and templates in this repository or if it's just better to use/develop these under [Azure Quickstart Templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) or [ESLZ Workload Templates](https://github.com/Azure/Enterprise-Scale/tree/main/workloads)

**Comment by AlexanderSehr** [comment]
## Workloads folder
I'd argue against the [ESLZ Workloads folder](https://github.com/Azure/Enterprise-Scale/tree/main/workloads) to have a clear separation of concerns. While the ESLZ repo could use the modules in this repository those would not need to be in the same. At the same time, other solutions can reference this repository's modules without confusing it with the ESLZ deployment itself.

## Quickstart
Regarding the [Quickstart templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules) I'd again argue for a dedicated solution (like this repository) as the Quickstart templates represent a huge library of useful templates for all kinds of scenarios. However, they are not maintained and from the structure it would not be clear where our `deploy.json` templates would be located. Though they follow the same structure as us with the resource provider, it is then just a long list of ARM templates. Furthermore, as .bicep shifts into focus it makes the situation even more confusing. 
Hence, I'd propose to keep this repository as it's own sanitized solution that provides flexible modules that can be leveraged for any kind of use case, while not being use-case specific at all.

**Reply by segraef** [reply]
Agreed. Thank you very much for your input @MrMCake and @ahmadabdalla.
